Can't run `byteBuddyJava` task twice without stopping Gradle daemon

```
 ./gradlew :instrumentation:jetty-8.0:bytebuddyjava --no-build-cache --stacktrace --rerun-tasks
 ./gradlew :instrumentation:jetty-8.0:bytebuddyjava --no-build-cache --stacktrace --rerun-tasks
```

fails with

```
Caused by: java.lang.RuntimeException: java.lang.IllegalStateException: Error reading class io.opentelemetry.javaagent.instrumentation.servlet.v3_0.TagSettingAsyncListener
        at io.opentelemetry.javaagent.tooling.muzzle.collector.MuzzleCodeGenerator$GenerateMuzzleReferenceMatcherMethodAndField.visitEnd(MuzzleCodeGenerator.java:458)
        at net.bytebuddy.dynamic.scaffold.TypeWriter$Default$ForInlining$WithFullProcessing$RedefinitionClassVisitor.onVisitEnd(TypeWriter.java:5140)
        at net.bytebuddy.utility.visitor.MetadataAwareClassVisitor.visitEnd(MetadataAwareClassVisitor.java:370)
        at net.bytebuddy.jar.asm.ClassReader.accept(ClassReader.java:715)
        at net.bytebuddy.jar.asm.ClassReader.accept(ClassReader.java:394)
        at net.bytebuddy.dynamic.scaffold.TypeWriter$Default$ForInlining.create(TypeWriter.java:3892)
        at net.bytebuddy.dynamic.scaffold.TypeWriter$Default.make(TypeWriter.java:2166)
        at net.bytebuddy.dynamic.scaffold.inline.RebaseDynamicTypeBuilder.make(RebaseDynamicTypeBuilder.java:252)
        at net.bytebuddy.build.Plugin$Engine$Default$Preprocessor$Resolved.call(Plugin.java:4498)
        at net.bytebuddy.build.Plugin$Engine$Default$Preprocessor$Resolved.call(Plugin.java:4455)
        at net.bytebuddy.build.Plugin$Engine$Dispatcher$ForSerialTransformation.accept(Plugin.java:3538)
        at net.bytebuddy.build.Plugin$Engine$Default.apply(Plugin.java:4302)
        at net.bytebuddy.build.gradle.AbstractByteBuddyTask.doApply(AbstractByteBuddyTask.java:331)
        ... 99 more
Caused by: java.lang.IllegalStateException: Error reading class io.opentelemetry.javaagent.instrumentation.servlet.v3_0.TagSettingAsyncListener
        at io.opentelemetry.javaagent.tooling.muzzle.collector.ReferenceCollector.collectReferencesFrom(ReferenceCollector.java:101)
        at io.opentelemetry.javaagent.tooling.muzzle.collector.MuzzleCodeGenerator$GenerateMuzzleReferenceMatcherMethodAndField.generateReferences(MuzzleCodeGenerator.java:144)
        at io.opentelemetry.javaagent.tooling.muzzle.collector.MuzzleCodeGenerator$GenerateMuzzleReferenceMatcherMethodAndField.visitEnd(MuzzleCodeGenerator.java:207)
        ... 111 more
Caused by: java.util.zip.ZipException: ZipFile invalid LOC header (bad signature)
        at net.bytebuddy.jar.asm.ClassReader.readStream(ClassReader.java:309)
        at net.bytebuddy.jar.asm.ClassReader.<init>(ClassReader.java:277)
        at io.opentelemetry.javaagent.tooling.muzzle.collector.ReferenceCollector.collectReferencesFrom(ReferenceCollector.java:83)
        ... 113 more
```

If I do `./gradlew --stop` and run the task, it succeeds, then run again and fails - `./gradlew --stop` always brings it back to a working state for one attempt until the next time the daemon is stopped.

--- a/javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/muzzle/collector/ReferenceCollector.java
+++ b/javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/muzzle/collector/ReferenceCollector.java
@@ -16,6 +16,7 @@ import io.opentelemetry.javaagent.tooling.Utils;
 import io.opentelemetry.javaagent.tooling.muzzle.Reference;
 import java.io.IOException;
 import java.io.InputStream;
+import java.net.URLConnection;
 import java.util.ArrayDeque;
 import java.util.ArrayList;
 import java.util.HashMap;
@@ -55,14 +56,7 @@ public class ReferenceCollector {
       String visitedClassName = instrumentationQueue.remove();
       visitedClasses.add(visitedClassName);
 
-      try (InputStream in =
-          checkNotNull(
-              ReferenceCollector.class
-                  .getClassLoader()
-                  .getResourceAsStream(Utils.getResourceName(visitedClassName)),
-              "Couldn't find class file %s",
-              visitedClassName)) {
-
+      try (InputStream in = getClassFileStream(visitedClassName)) {
         // only start from method bodies for the advice class (skips class/method references)
         ReferenceCollectingClassVisitor cv = new ReferenceCollectingClassVisitor(isAdviceClass);
         ClassReader reader = new ClassReader(in);
@@ -89,6 +83,24 @@ public class ReferenceCollector {
         isAdviceClass = false;
       }
     }
+  }
+
+  private static InputStream getClassFileStream(String className) throws IOException {
+    URLConnection connection =
+        checkNotNull(
+                ReferenceCollector.class
+                    .getClassLoader()
+                    .getResource(Utils.getResourceName(className)),
+                "Couldn't find class file %s",
+                className)
+            .openConnection();
+
+    // Since the JarFile cache is not per class loader, but global with path as key, using cache may
+    // cause the same instance of JarFile being used for consecutive builds, even if the file has
+    // been changed. There is still another cache in ZipFile.Source which checks last modified time
+    // as well, so the zip index is not scanned again on every class.
+    connection.setUseCaches(false);
+    return connection.getInputStream();
   }
 
   private void addReference(String refClassName, Reference reference) {

